### PR TITLE
Add option to set up the distance matrix used, with a possible squashing factor.

### PR DESCRIPTION
**Pull request by sarahlou-dr**
Set up the distance matrix with distances between the different number tokens in the vocabulary, a feature only relevant for the Wasserstein-based NTLoss. 
The relative difference between the distance to the furthest digit and the closest digit can be tuned using the introduced squashing factor.

### CODE REVIEW

File: ntloss/core.py [hunk 1]
```diff
@@ -234,9 +309,12 @@ def forward(
         softmax_probs = F.softmax(nt_logits, dim=-1)
 
         # compute absolute difference between the true numbers and all possible number values
-        abs_diff = torch.abs(
-            y[valid_positions].unsqueeze(-1) - self.number_values_dense
-        )
+        if self.setup_dist_lookup or isinstance(self.squash_factor, Number):
```
Comment (jannisborn) [inline ntloss/core.py:312]: here we can then always do the dist lookup. We just need to ensure the devices match, so we have to cast both attributes to the device on which labels live

File: tests/test_ntloss.py [hunk 2]
```diff
@@ -136,3 +136,94 @@ def test_correct_minimum(loss_class, logit_builder):
...
+
+    assert not torch.isnan(losses_lookup).any(), "Encountered NaN in loss matrix"
+
+    assert torch.allclose(losses, losses_lookup, atol=1e-8), (
+        "Loss should be the same whether lookup table is used or not, as long as squashing ",
+        "factor is not set.",
+    )
+
+
+@pytest.mark.parametrize("loss_class", [NTLoss])
```
Comment (jannisborn) [inline tests/test_ntloss.py:179]: here it would be nice to add a test with an exotic vocab like {0, 2, 4, 8}

File: ntloss/core.py [hunk 3]
```diff
@@ -11,39 +12,58 @@
...
+                    ):
+                        self.number_values[id] = float(token)
+                else:
                     self.number_values[id] = float(token)
 
         self.is_number_token = ~torch.isnan(self.number_values)
         self.number_values_dense = self.number_values[self.is_number_token]
 
+        if self.digit_level:
+            assert len(self.number_values_dense) == 10, "If digit level "
```
Comment (jannisborn) [inline ntloss/core.py:65]:
```suggestion
            assert len(self.number_values_dense) == 10, f"You requested digit-level but more than 10 number tokens were identified: {self.number_values_dense}"
```
